src/components/WorkQueueCombobox.vue

```diff
@@ -49,20 +51,68 @@
   const api = useWorkspaceApi()
   const workQueuesSubscription = useSubscription(api.workQueues.getWorkQueues, [{}])
   const workQueues = computed(() => workQueuesSubscription.response ?? [])
-  const options = computed<SelectOption[]>(() => {
-    const options: SelectOption[] = workQueues.value.map(workQueue => ({
-      // Any consumers of the work queue should subscribe to it by name and not id
-      value: workQueue.name,
-      label: workQueue.name,
-    }))
+  const options = computed<SelectOptionGroup[]>(() => {
+    const workQueuesGroupedByWorkPool = workQueues.value.reduce<Map<string, WorkQueue[]>>((acc, workQueue) => {
+      // Filter options by work pool id or name
+      if (props.workPoolFilter?.id || props.workPoolFilter?.name) {
+        if (!props.workPoolFilter.id?.includes(workQueue.workPoolId) && !(workQueue.workPoolName && props.workPoolFilter.name?.includes(workQueue.workPoolName))) {
+          return acc
+        }
+      }
```

Comment:
Couldn't we just pass `props.workPoolFilter` into the `getWorkQueues`? Then this options computed would just group and create options for whatever work queues are returned from the api.

Reply:
That api doesn't support filtering by work pools? 

so I'm filtering client-side

Reply:
hmm we should probably update the api rather than filtering client side. As is it looks like this would support anything in the WorkPoolFilter (like `type`). 

If updating the api isn't an option maybe just add props for `workPoolName` or `workPoolId` so that way its clear what the component supports filtering by.

Reply:
Hm. how about just warning if type is provided? 

server-side updates aside, I think I'd rather have this singular object interface for filtering the workpool over more discrete props and then I grabbed the WorkPoolFilter b/c it's semantically correct and the reuse seemed nice for familiarity rather than creating some other structure for like { id?: string[], name?: string[] }... 🤔 but maybe I just do that and lean into structural typing... like a consumer could still pass a WorkPoolFilter 🤷

Reply:
I don't feel strongly other than making it type safe. If we ever add something to WorkPoolFilter and then this component doesn't work as expected because it accepts a WorkPoolFilter but doesn't implement the whole thing is  a land mind for a bug IMO. I'd prefer to update the api to allow filtering by workPool as that's the safest way to ensure things will continue to work properly. But I know you're down a rabbit hole already. So just making this component type safe would be my bar.

Reply:
Hm yeah +1 more narrowly accurate structural types > nominal typing.

I'm gonna switch it to its own little id/name type. 

it probably is a little off b/c the filters are generally like "api filters" even if they're not labeled as such... so I see where this works out better if the filtering was server-side and this prop is just going straight into api filters. 👍

Reply:
yeah pretty much every (maybe all) "ThingFilter" is a type for the api and we don't usually use the term "filter" for client side stuff. But it be fine as long as the client side enforced the whole filter was implemented or could accurately error for things that were not implemented. But easier and safer to just narrow the type to exactly what you support. 

Thanks for talking that through!

Reply:
Can you not use the getWorkPoolQueues route to filter by work pool?

Reply:
The work pool queue apis don't support "breaking out" of a single work pool.

This - without like a "get all work pools -> for each work pool: get all work queues"
<img width="796" alt="image" src="https://github.com/PrefectHQ/prefect-ui-library/assets/22418768/6a3208d5-72a2-43c2-8b60-ea77b6e19c2d">

Reply:
> I'm gonna switch it to its own little id/name type.

addressed in 00b3d75